src/helics/core/ActionMessage.cpp

```diff
@@ -275,12 +277,48 @@ int ActionMessage::serializedByteCount() const
 std::string ActionMessage::to_string() const
 {
     std::string data;
-    auto sz = serializedByteCount();
-    data.resize(sz);
-    toByteArray(&(data[0]), sz);
+    if (checkActionFlag(*this, use_json_serialization_flag)) {
+        data = to_json_string();
+    } else {
+        auto sz = serializedByteCount();
+        data.resize(sz);
+        toByteArray(&(data[0]), sz);
+    }
     return data;
 }
 
+std::string ActionMessage::to_json_string() const
+{
+    Json::Value packet;
+    packet["version"] =
+        HELICS_VERSION_MAJOR * 10000 + HELICS_VERSION_MINOR * 100 + HELICS_VERSION_PATCH;
```

Comment:
What will the version be used for? Is it something that could just be sent once when the connection is first established?

Reply:
We need some sort of version indicator to match up future changes to the json output. I don't think we could only send it once, otherwise we would have to record it and send it to the interpreter every time.    Right now it isn't used since there is only one version,  but it would start to get used if we ever upgrade the message format.